cli/openshift.go

```diff
@@ -0,0 +1,57 @@
+package cli
+
+import (
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+type OpenShiftOptions struct {
+	Namespace string
+	Labels    map[string]string
+}
+
+type OpenshiftReport struct {
+	Stdout string
+	Stderr string
+	Items  []OperatorSdkScorecardItem `json:"items"`
+}
+
+type SubscriptionData struct {
+	Name                   string
+	Channel                string
+	CatalogSource          string
+	CatalogSourceNamespace string
+	Package                string
+}
+
+type CatalogSourceData struct {
+	Name  string
+	Image string
+}
+
+type OperatorGroupData struct {
+	Name             string
+	TargetNamespaces []string
+}
+
+type OpenshiftEngine interface {
+	CreateNamespace(name string, opts OpenShiftOptions, config *rest.Config) (*OpenshiftReport, error)
```

Comment:
Here's my feeling on this. Tying the interface to rest.Config makes it less useful as an interface. It seems like that should either be included in OpenShiftOptions, or wrapped in something more "generic".  Otherwise, there will only ever really be a single implementation of this interface, thereby rendering it redundant and unnecessary.

Reply:
OpenshiftOptions contains the info for the resource creation (such as namespace and labels). IMO using `opts` to pass along the *rest.Config is dangerous as the client does not need this and the "least privileged principle" will be violated.

Passing `*rest.config` to a method will give the clients the flexibility to communicate with any k8s cluster.

Reply:
I don't see that we would have a reason to have any other implementation of this than the one that is here. It may be the subject of a refactoring down the road.